manta-pay/src/signer/base.rs

```diff
@@ -54,15 +54,26 @@ pub struct HierarchicalKeyDerivationFunction<C = Testnet>(PhantomData<C>)
 where
     C: CoinType;
 
-impl<C> KeyDerivationFunction for HierarchicalKeyDerivationFunction<C>
+impl<C> SecretKeyType for HierarchicalKeyDerivationFunction<C>
 where
     C: CoinType,
 {
-    type Key = <KeySecret<C> as HierarchicalKeyDerivationScheme>::SecretKey;
-    type Output = SecretKey;
+    type SecretKey = <KeySecret<C> as HierarchicalKeyDerivationScheme>::SecretKey;
+}
 
+impl<C> PublicKeyType for HierarchicalKeyDerivationFunction<C>
+where
+    C: CoinType,
+{
+    type PublicKey = SecretKey;
+}
+
+impl<C> Derive for HierarchicalKeyDerivationFunction<C>
```

Comment:
This is not technically a key agreement scheme so it should not use this API, it should use the other KDF one.